assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Analytics-4 Settings controls.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useState, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { FORM_SETUP, MODULES_ANALYTICS } from '../../datastore/constants';
+import { Select, Option } from '../../../../material-components';
+import { GA4ActivateSwitch } from '../common';
+import {
+	PropertySelect,
+	UseSnippetSwitch,
+} from '../../../analytics-4/components/common';
+const { useSelect, useDispatch } = Data;
+
+export default function GA4SettingsControls() {
+	const [ matchedProperty, setMatchedProperty ] = useState();
+
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	// This select is needed to check whether the AdminAPI works or not.
+	useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getProperties( accountID )
+	);
+
+	const isAdminAPIWorking = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isAdminAPIWorking()
+	);
+
+	const enableGA4 = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'enableGA4' )
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const { matchAccountProperty, setPropertyID } = useDispatch(
+		MODULES_ANALYTICS_4
+	);
+
+	useEffect( () => {
+		setMatchedProperty( undefined );
+		matchAccountProperty( accountID ).then( setMatchedProperty );
+	}, [ accountID, setMatchedProperty, matchAccountProperty ] );
```

Comment:
See my other comment in https://github.com/google/site-kit-wp/pull/4229#discussion_r741547616, I think we should also account for `isAdminAPIWorking` here and only trigger the logic if it does work.

Reply:
@felixarntz, I have added a check to the `useEffect` hook to make sure `isAdminAPIWorking` is true. I like this approach better than moving this logic to the `onActivate` callback because if we do it, there will be a lag between actual activation and the end of the property matching process which is worse user experience.

Reply:
Keeping this in `useEffect` sounds good!